src/hotspot/share/prims/jvm.cpp

```diff
@@ -947,10 +947,10 @@ static jclass jvm_lookup_define_class(jclass lookup, const char *name,
 
   InstanceKlass* ik = NULL;
   if (!is_hidden) {
-    ik = SystemDictionary::resolve_from_stream(class_name,
+    ClassLoadInfo cl_info(protection_domain);
```

Comment:
Minor comment, you could pull the creation of ClassLoadInfo out of this if statement since both the the if and the else sections create a ClassLoadInfo with pretty much the same information.

Reply:
That other ClassLoadInfo cl_info(protection_domain) you see is from another function, so I can't pull it out.
The other side of the 'if' statement creates a ClassLoadInfo with all the hidden class goodies.

Reply:
In jvm_lookup_define_class there are 2 ClassLoadInfo cl_info constructions on line #950 and line #961 that could be pull out of the if statement.  Again comment is minor and I am ok with how you decide to go forward with it.